defold-spine/commonsrc/vertices.cpp

```diff
@@ -35,6 +37,89 @@ static uint32_t EnsureArrayFitsNumber(dmArray<T>& array, uint32_t num_to_add)
     return prev_size;
 }
 
+
+void GetSkeletonBounds(const spSkeleton* skeleton, SpineModelBounds& bounds)
+{
+    dmArray<float> scratch; // scratch buffer
+
+    // a "negative" bounding rectangle for starters
+    bounds.minX = FLT_MAX;
+    bounds.minY = FLT_MAX;
+    bounds.maxX = FLT_MIN;
+    bounds.maxY = FLT_MIN;
+
+    // For each slot in the draw order array of the skeleton
+    for (int s = 0; s < skeleton->slotsCount; ++s)
+    {
+        spSlot* slot = skeleton->drawOrder[s];
+
+        // Fetch the currently active attachment, continue
+        // with the next slot in the draw order if no
+        // attachment is active on the slot
+        spAttachment* attachment = slot->attachment;
+        if (!attachment)
+        {
+            continue;
+        }
+
+        // Fill the vertices array depending on the type of attachment
+        if (attachment->type == SP_ATTACHMENT_REGION)
+        {
+            // Cast to an spRegionAttachment so we can get the rendererObject
+            // and compute the world vertices
+            spRegionAttachment* regionAttachment = (spRegionAttachment*)attachment;
+
+            EnsureArrayFitsNumber(scratch, 4*2);
+
+            // Computed the world vertices positions for the 4 vertices that make up
+            // the rectangular region attachment. This assumes the world transform of the
+            // bone to which the slot (and hence attachment) is attached has been calculated
+            // before rendering via spSkeleton_updateWorldTransform
+            spRegionAttachment_computeWorldVertices(regionAttachment, slot->bone, scratch.Begin(), 0, 2);
+
+            // go through vertex coords and update max/min for X and Y
+            for (int i=0; i<4*2; i+=2) {
+                bounds.minX = dmMath::Min(scratch[i], bounds.minX);
```

Comment:
Due to aliasing of the `scratch` pointer, I expect this loop to be somewhat faster if you store the x/y in temp variables:
```
float x = *scratch++;
float y = *scratch++;
bounds.minX = dmMath::Min(x, bounds.minX);
bounds.maxX = dmMath::Max(x, bounds.maxX);
...
```
```

Reply:
hmm, `scratch` is defined as `dmArray<float> scratch`. 

I guess you meant something like:

```
float* coords = scratch.Begin();
for (int i=0; i<4*2; i+=2) {
    float x = *coords++;
    float y = *coords++;
    bounds.minX = dmMath::Min(x, bounds.minX);
    bounds.minY = dmMath::Min(y, bounds.minY);
    bounds.maxX = dmMath::Max(x, bounds.maxX);
    bounds.maxY = dmMath::Max(y, bounds.maxY);
}
```

or am i missing something ?

Reply:
Correct.